app/javascript/components/firmware-registry-add-edit.jsx

```diff
@@ -0,0 +1,135 @@
+    });
+  };
+
+  const handleFormStateUpdate = (formState) => {
+    props.dispatch({
+      type: 'FormButtons.saveable',
+      payload: formState.valid,
+    });
+    props.dispatch({
+      type: 'FormButtons.pristine',
+      payload: formState.pristine,
+    });
+    props.dispatch({
+      type: 'FormButtons.callbacks',
+      payload: { addClicked: () => submitValues(formState.values) },
+    });
+  };
+
+  return (
+    <Grid fluid>
+      <Row>
+        <MiqFormRenderer
+          initialValues={initialValues}
+          schema={
+              {
+                fields: [
+                  {
+                    component: componentTypes.SELECT_COMPONENT,
+                    name: 'type',
+                    options: [{ value: 'FirmwareRegistry::RestApiDepot', label: __('Rest API Depot') }],
+                    label: __('Registry Type'),
+                    placeholder: __('Nothing selected'),
+                    isSearchable: true,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    validateOnMount: true,
+                    isDisabled: true,
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'name',
+                    label: __('Name'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'url',
+                    label: __('URL'),
+                    placeholder: 'https://example.com:1234/images',
+                    maxLength: 2048,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'userid',
+                    label: __('Username'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'password',
+                    label: __('Password'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    type: 'password',
+                  },
+                ],
+              }
+          }
+          onSubmit={() => {}}
```

Comment:
You should always have onSubmit callback. I don't see any changes to form values while passing it to the redux. You can the same callback here. There other ways to submit a form than just the buttons.

Reply:
Okaaaay. This callback was never called when I was testing it, but I guess parent component prevents bubbling event in case of popup, right?